cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/Grant.java

```diff
@@ -83,20 +82,15 @@
  * @author majorb
  *
  */
-public class Grant {
+public abstract class Grant {
 
     private final URI artifactURI;
-    
-    public Date releaseDate;
-    public boolean isPublic = false;
-    public final Date evaluationDate = new Date();
-    
-    private final List<GroupURI> readGroups = new ArrayList<GroupURI>();
-    private final List<GroupURI> readWriteGroups = new ArrayList<GroupURI>();
+    protected final List<GroupURI> groups = new ArrayList<GroupURI>();
+    protected final Date expiryDate = new Date();
```

Comment:
expireDate cannot be final and set to "now"... different sources of permissions will decide on the expiryDate. If it is mandatory (I think it should be) then the field should be private and it should be included in the constructor args.

Reply:
Agree it should be mandatory, can always be a large value if it's not applicable for some reason. Added to the constructor.